src/memory.rs

```diff
@@ -1,48 +1,109 @@
+use candid::Principal;
 use ic_stable_structures::memory_manager::{MemoryId, MemoryManager, VirtualMemory};
 #[cfg(target_arch = "wasm32")]
 use ic_stable_structures::DefaultMemoryImpl;
+use ic_stable_structures::StableBTreeMap;
 #[cfg(not(target_arch = "wasm32"))]
 use ic_stable_structures::VectorMemory;
-use ic_stable_structures::{Cell, StableBTreeMap};
 use std::cell::RefCell;
 
-use crate::{
-    constants::NODES_IN_FIDUCIARY_SUBNET,
-    types::{AuthSet, Metadata, Metrics, PrincipalStorable, Provider, StorableRpcService},
-};
+use crate::types::{ApiKey, Metrics, PrincipalStorable, ProviderId};
 
 #[cfg(not(target_arch = "wasm32"))]
 type Memory = VirtualMemory<VectorMemory>;
 #[cfg(target_arch = "wasm32")]
 type Memory = VirtualMemory<DefaultMemoryImpl>;
 
 thread_local! {
-    // Unstable static data: this is reset when the canister is upgraded.
+    // Unstable static data: these are reset when the canister is upgraded.
     pub static UNSTABLE_METRICS: RefCell<Metrics> = RefCell::new(Metrics::default());
-    static UNSTABLE_SUBNET_SIZE: RefCell<u32> = RefCell::new(NODES_IN_FIDUCIARY_SUBNET);
+    static UNSTABLE_DEMO_STATUS: RefCell<bool> = RefCell::new(false);
 
-    // Stable static data: this is preserved when the canister is upgraded.
+    // Stable static data: these are preserved when the canister is upgraded.
     #[cfg(not(target_arch = "wasm32"))]
-    pub static MEMORY_MANAGER: RefCell<MemoryManager<VectorMemory>> =
+    static MEMORY_MANAGER: RefCell<MemoryManager<VectorMemory>> =
         RefCell::new(MemoryManager::init(VectorMemory::new(RefCell::new(vec![]))));
     #[cfg(target_arch = "wasm32")]
-    pub static MEMORY_MANAGER: RefCell<MemoryManager<DefaultMemoryImpl>> =
+    static MEMORY_MANAGER: RefCell<MemoryManager<DefaultMemoryImpl>> =
         RefCell::new(MemoryManager::init(DefaultMemoryImpl::default()));
-    pub static METADATA: RefCell<Cell<Metadata, Memory>> = RefCell::new(Cell::init(
-            MEMORY_MANAGER.with(|m| m.borrow().get(MemoryId::new(0))),
-            Metadata::default()).unwrap());
-    pub static AUTH: RefCell<StableBTreeMap<PrincipalStorable, AuthSet, Memory>> = RefCell::new(
-        StableBTreeMap::init(MEMORY_MANAGER.with(|m| m.borrow().get(MemoryId::new(1)))));
-    pub static PROVIDERS: RefCell<StableBTreeMap<u64, Provider, Memory>> = RefCell::new(
-        StableBTreeMap::init(MEMORY_MANAGER.with(|m| m.borrow().get(MemoryId::new(2)))));
-    pub static SERVICE_PROVIDER_MAP: RefCell<StableBTreeMap<StorableRpcService, u64, Memory>> = RefCell::new(
-        StableBTreeMap::init(MEMORY_MANAGER.with(|m| m.borrow().get(MemoryId::new(3)))));
+    static API_KEY_MAP: RefCell<StableBTreeMap<ProviderId, ApiKey, Memory>> =
+        RefCell::new(StableBTreeMap::init(MEMORY_MANAGER.with_borrow(|m| m.get(MemoryId::new(5)))));
```

Comment:
We originally used IDs 0-3 for providers, authorizations, etc. in the previous version. Although this wouldn't make a difference in mainnet if we reinstall the canister, it could affect local replica deployments in ways that are potentially difficult to debug. At one point I also had something using `4` but it was also removed at some point.

I'd be fine changing the IDs to 0 and 1 if you think this wouldn't cause any significant issues.